File: tree/0538_convert_bst_to_greater_tree/0538_convert_bst_to_greater_tree.go
package main

import "fmt"

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func convertBst(root *TreeNode) *TreeNode {
	sum := 0
	travel(root, &sum)
	return root
}

func travel(root *TreeNode, sum *int) {
	if root == nil {
		return
	}
	travel(root.Right, sum)
	*sum += root.Val
	root.Val = *sum
	travel(root.Left, sum)
}

func main() {
	root := &TreeNode{5, nil, nil}
	root.Left = &TreeNode{2, nil, nil}
	root.Right = &TreeNode{13, nil, nil}
	res := convertBst(root)
	fmt.Println(res.Left.Val)
}

File: tree/0538_convert_bst_to_greater_tree/0538_convert_bst_to_greater_tree.py
class TreeNode(object):
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None


class Solution(object):
    def convertBST(self, root):
        def covnertBSTHelper(root, cur_sum):
            if not root:
                return cur_sum
            if root.right:
                cur_sum = covnertBSTHelper(root.right, cur_sum)
            cur_sum += root.val
            root.val = cur_sum
            if root.left:
                cur_sum = covnertBSTHelper(root.left, cur_sum)
            return cur_sum
        covnertBSTHelper(root, 0)
        return root


root = TreeNode(5)
root.left = TreeNode(2)
root.right = TreeNode(13)
res = Solution().convertBST(root)
print(res.left.val)